### PR TITLE
Feature/80   student routing

### LINKED ISSUE #80: [FEATURE] - Update routing permissions to now consider the student role
### Reference Issues

_No response_

### Summary

Update of routing permissions for student role

### Unresolved questions

_No response_

### Definition of Done

When all routes have been considered for new permission.
A variable on the frontend server to determine user role from JWT token

### PR DESCRIPTION
**Pull request by Schoogle**
## Description

This pull request adds role based routing restrictions

### Resolved Issue
Fixes #80 

## Changes

We now extract the role from the jwt token, only after it has been validated buy the backend API.
Allowed routes are updated to reflect the roles
Fixed login not displaying backend error, if zod validation did not catch it

Current tests have been changed to include a student role on user objects. Tests including the instructor class should be considered

## Additional Information

Index page is now shown differently based on if you are a instructor or not. Perhaps the instructor home should be moved to its own route and not be on the main index.

## Checklist

- [X] I have added test cases for my additions - Tests for what routes are allowed for the roles are not fully implemented
- [X] New and old tests passed
- [X] Documentation is updated

### CODE REVIEW

File: __tests__/integration/routes/+layout.test.ts [hunk 1]
```diff
@@ -45,9 +45,9 @@ describe('get_anon_userID', () => {
 	});
 
 	it('should redirect authorized user when accessing login url ', () => {
-		const user = { name: 'Andreas' };
+		const user = { name: 'Andreas', role: "Student" };
```
Comment (andreaswurtz) [inline __tests__/integration/routes/+layout.test.ts:48]: role: "King"

File: __tests__/unit/lib/getrole.test.ts
Comment (andreaswurtz) [inline __tests__/unit/lib/getrole.test.ts:1]: Øhh you are not testing when it fails???????

File: src/routes/signup/+page.server.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ export const actions: Actions = {
       });
     } else {
       const error = await response.json();
-      return setError(form, 'confirmPassword', error?.detail || 'Signup failed!');
+      return setError(form, 'confirmPassword', error.errors.Password.join('\n') || "Signup failed");
```
Comment (andreaswurtz) [inline src/routes/signup/+page.server.ts:41]: meget bedre

**Review by andreaswurtz — APPROVED** [review]
lgtm

File: __tests__/unit/lib/getrole.test.ts
Comment (Schoogle) [reply __tests__/unit/lib/getrole.test.ts:1]: it is now tested

File: __tests__/integration/routes/+layout.test.ts (hunk 1 above)
Comment (Schoogle) [reply __tests__/integration/routes/+layout.test.ts:48]: i do not appreciate your patetic attempt at a clever jest

**Review by andreaswurtz — APPROVED** [review]
lgbt

**Pull request merged**